tests/test_bfs_multihop.py

```
@@ -692,10 +692,48 @@ def test_scoredhop_dataclass_shape() -> None:
     assert h.score == 0.5
     assert h.depth == 1
     assert h.path == [EDGE_SUPPORTS]
+    # belief_id_trail defaults to empty for backwards-compat constructors.
+    assert h.belief_id_trail == ()
     with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)